cmd/kots/cli/metrics.go

```diff
@@ -0,0 +1,202 @@
+	im.FinishedAt = time.Now()
+	url := fmt.Sprintf("%s/kots_metrics/finish_install/%s", upstream, im.InstallID)
+	return PostMetrics(url, im)
+}
+
+//PostMetrics does HTTP to post metrics
+func PostMetrics(url string, im *InstallMetrics) error {
+	log := logger.NewLogger()
+	// Marshall im and POST metrics to upstream
+	jsonStr, _ := json.Marshal(*im)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
+	req.Header.Set("Content-Type", "application/json")
+	// 1sec timeout to reach the server, dont wait for too long
+	client := &http.Client{
+		Timeout: time.Second * 1,
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Error(err)
+		return errors.Wrap(err, "Request failed")
+	}
+	_, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Error(err)
+		return errors.Wrap(err, "ReadAll failed")
+	}
+	defer resp.Body.Close()
+	// Do a Post to the endpoint present in the license
+	return nil
+}
+
+// InitMetrics initializes with static data
+func InitMetrics(deployOptions *kotsadmtypes.DeployOptions, isAirgap bool) (*KotsInstallMetrics, error) {
+
+	// Dont do metrics in airgap as there is no outbound n/w
+	if isAirgap == true {
+		return nil, nil
+	}
+	m := &KotsInstallMetrics{}
+	im := &m.install
+	if deployOptions.License != nil {
+		im.AppSlug = deployOptions.License.Spec.AppSlug
+	} else {
+		// if the license file is not passed, use the namespace
+		im.AppSlug = deployOptions.Namespace
+	}
+	m.endpoint.UpstreamEp = getEndpoint(deployOptions)
+	im.StartedAt = time.Now()
+	// TODO Get the MachineID and use it as ClusterID to check pre and post upgrades across different clusters
+	kubectl, err := exec.LookPath("kubectl")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find kubectl")
+	}
+	out, err := exec.Command(kubectl, "get", "ns", "kube-system", "-o", "jsonpath='{.metadata.uid}'").Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find kubectl")
+	}
+	_ = string(out[:])
+	// machineID := string(out[:])
+
+	// use guid as InstallID
+	im.InstallID = ksuid.New().String()
+
+	// Get the Host OS
+	dist, distVersion, _ := discover()
+	im.HostOS = dist + " " + distVersion
+	im.KotsVersion = version.Version()
+	return m, nil
+}
+
+//ProductionEp present in license file: https://replicated.app
+//codeServerSvc/Dev env present in license file: file https://replicated-app
+//codeServerEP use this if no license file is passed http://localhost:30016
+func getEndpoint(deployOptions *kotsadmtypes.DeployOptions) string {
+
+	cliConfigPath := "src/github.com/replicatedhq/kots/cmd/kots/cli/config"
+	// if the license file is passed as argument, and if the endpoint is not a dev endpoint
+	if deployOptions.License != nil {
+		if isDevEndpoint(deployOptions.License.Spec.Endpoint) == false {
+			return deployOptions.License.Spec.Endpoint
+		}
+	}
+	// if license file not passed, pick the dev endpoint from the dev-config file
+	devCfgFile := fmt.Sprintf("%s/%s/dev-cfg.json", os.Getenv("GOPATH"), cliConfigPath)
```

Comment:
this should  be in an environment variable instead of a file

Reply:
As we discussed, we can keep this file to avoid punching development metrics into the production endpoints